docs/UserGuide.md

```diff
@@ -39,136 +39,281 @@ AddressBook Level 3 (AB3) is a **desktop app for managing contacts, optimized fo
 
-### Clearing all entries : `clear`
+#### What to Expect
+- **If Successful:** The details of the customer with the specified ID will be displayed.
+- **If There is an Error:** 
+  - Invalid ID: "No customer with `<Customer ID>` exists. Please recheck the ID."
 
-Clears all entries from the address book.
+**Handling Duplicates:**  
+- There can be no duplicate customer IDs due to system constraints on ID uniqueness.
+- While other information like name and address can be duplicated, each customer ID is unique, ensuring you always retrieve the correct customer record.
 
-Format: `clear`
+---
+### Feature 5: Find a Customer by Details
+
+**Purpose:**  
+Enables users to search for and display all customers who match the specified details such as name, job, or ID.
+
+**How to Use It:**  
+- **Command Format:** 
+```
+filter --<flag> <flag field>
+```
+- **Examples:** 
+```
+filter --name TAN LESHEW
+filter --job doctor of computer science
+filter --id 69
+```
+
+#### Parameters
+| Parameter | Expected Format       | Explanation                                                                 |
+|-----------|-----------------------|-----------------------------------------------------------------------------|
+| name      | Any string            | Input will be converted to block letters for consistent and accurate search.|
+| job       | Any string            | Searches are case-insensitive, reflecting the non-sensitive nature of job titles.|
+| id        | Integer number        | Must be an integer as IDs are unique identifiers for each customer.         |
+
+#### What to Expect
+- **If Successful:** 
+  - Message: "Here are all the customers that match your search: (List of customers)."
+- **If There is an Error:** 
+  - Missing field: "Please provide a name/job/id." (Specific to the flag used)
+  - Invalid or unsupported flag: "Please provide a valid flag."
+  - No matching customers: "No customers were found. Please check the name/job/id." (Specific to the flag)
+
+**Handling Duplicates:**  
+- For `name` and `job` fields: Multiple customers can have the same names or jobs. The command will list all matching entries.
+- For the `id` field: Since customer IDs are intended to be unique, finding multiple customers with the same ID will
+  trigger a warning: 
+  - "There seems to be multiple customers tagged to the same ID, use the delete command to remove all customers with the affected ID, and re-add customers accordingly."
+---
 
-### Exiting the program : `exit`
+### Feature 6: Save Remarks About Customers
+
+**Purpose:**  
+Allows users to save specific notes or remarks about a customer, which can be viewed later to recall notable details.
+
+**How to Use It:**  
+- **Command Format:** 
+```
+note <customer id> <note here>
+```
+- **Example:** 
+```
+note 55 He is a problematic customer.
```

Comment:
I would suggest changing `note` to `remark` so we don't have to refactor the command.